lib/vault.ts

```diff
@@ -191,6 +196,154 @@ export async function listAllSubmissions(
   }
 }
 
+/**
+ * This method returns a list of selected form submission records.
+ * The list contains the actual submission data
+ * @param formID - The form ID from which to retrieve responses
+ */
+export async function retrieveSubmissions(
+  ability: UserAbility,
+  formID: string,
+  ids: string[]
+): Promise<VaultSubmission[]> {
+  // Check access control first
+  try {
+    await checkAbilityToAccessSubmissions(ability, formID);
+  } catch (e) {
+    if (e instanceof AccessControlError)
+      logEvent(
+        ability.userID,
+        {
+          type: "Form",
+          id: formID,
+        },
+        "AccessDenied",
+        `Attempted to retrieve responses for form ${formID}`
+      );
+    throw e;
+  }
+
+  try {
+    let accumulatedResponses: VaultSubmission[] = [];
+    const documentClient = connectToDynamo();
+
+    let keys = ids.map((id) => {
+      return { FormID: formID, NAME_OR_CONF: `NAME#${id.trim()}` };
+    }) as unknown as KeysAndAttributes["Keys"];
```

Comment:
Now that it uses the `lib-dynamodb` lib I think you should not need `as unknown as KeysAndAttributes["Keys"]` anymore.

Reply:
Still needed a type otherwise line 275 where we re-assign the unprocessedkeys to keys complains. ` as Record<string, string>[]` did the trick.